core/Plugin/Report.php

```diff
@@ -231,11 +232,23 @@ protected function init()
      * might depend on a setting (such as Ecommerce) of a site. In such a case you can perform any checks and then
      * return `true` or `false`. If your report is only available to users having super user access you can do the
      * following: `return Piwik::hasUserSuperUserAccess();`
+     *
+     * Classes that override this method must make sure to call the parent version as well.
+     *
      * @return bool
      * @api
      */
     public function isEnabled()
     {
+        if ($this->isRequiresProfilableData()
+            || (isset($this->dimension) && $this->dimension->isRequiresProfilableData())
+        ) {
+            $isProfilable = Request::isCurrentPeriodProfilable();
+            if (!$isProfilable) {
+                return false;
```

Comment:
fyi haven't looked too much through the code yet and not tested yet but I would assume this will be responsible to remove certain reports from the reporting UI page when the data is not profilable? I suppose when changing the date the reports won't change since we wouldn't reload the available reports. to be checked. just thought I leave the comment here while I have it on my mind.

Reply:
I'll test this, it's been a while so I can't remember if this is something I tested or not.

Reply:
Actually if we don't remove reports/segments but disable them some other way, this changes a lot, so I'll wait for an answer there.

Reply:
@mattab any thoughts? For reports I would say we always show them and simply show a message why a report doesn't show any data (because not profilable).

For segments I think @mattab wanted segments to be removed (or someone else). But they may still be needed for historical data etc. Or if sometimes you have profilable data and sometimes not (we used to have this case on matomo org but might be bit edge case)

Reply:
Segments are removed in the UI but nowhere else. We can also disable them and provide a tooltip saying why it doesn't make sense to use them on the current period. But when creating the segment, the period is arbitrary (eg, maybe they want it to be applied to a period w/ profilable data). Seems odd to see it greyed out and realize you have to change the period to be able to use it.

Perhaps it would be better to detect when a segment in use (ie, in the `segment=` query parameter) isn't going to provide correct data because the current period is not profilable, and display a notification.

Reply:
@mattab can you provide an answer/opinion on ^?

Reply:
showing all reports but when there's no profilable data for the selected period, there is a footer message saying so